src/190B20.c

```diff
@@ -1542,15 +1772,559 @@ ActorPart* get_actor_part(Actor* actor, s32 partIndex) {
+                decorationTable->unk_750 = 0;
+                decorationTable->unk_764 = 0;
+                decorationTable->unk_768 = 0;
+                decorationTable->unk_7D8 = 0;
+                decorationTable->unk_7D9 = 0;
+
+                for (j = 0; j < ARRAY_COUNT(decorationTable->posX); j++) {
+                    decorationTable->posX[j] = partnerActor->currentPos.x;
+                    decorationTable->posY[j] = partnerActor->currentPos.y;
+                    decorationTable->posZ[j] = partnerActor->currentPos.z;
+                }
+
+                decorationTable->unk_7DA = 3;
+                decorationTable->unk_7DB = 0;
+                decorationTable->effectType = 0;
+
+                for (j = 0; j < ARRAY_COUNT(decorationTable->unk_8B0); j++) {
+                    decorationTable->unk_8B0[j] = NULL;
+                    decorationTable->decorationType[j] = 0;
+                }
+            }
+
+            if (part->flags >= 0) {
+                part->movement = heap_malloc(sizeof(*part->movement));
+                ASSERT(part->movement != NULL);
+            }
+
+            part->animationRate = 1.0f;
+            part->currentAnimation = 0;
+            part->unk_84 = -1;
+
+            if (part->idleAnimations != NULL) {
+                part->currentAnimation = func_80265CE8(part->idleAnimations, 1);
+                part->unk_84 = spr_load_npc_sprite(part->currentAnimation | 0x80000000, NULL);
+            }
+
+            if (i + 1 >= partCount) {
+                part->nextPart = NULL;
+                continue;
+            }
+
+            part->nextPart = heap_malloc(sizeof(*part->nextPart));
+            part = part->nextPart;
+            if (part == NULL) {
+                PANIC();
+            }
 
-INCLUDE_ASM(s32, "190B20", load_partner_actor);
+            part->nextPart = NULL;
+        }
+
+        partnerActor->shadow = create_shadow_type(0, partnerActor->currentPos.x, partnerActor->currentPos.y, partnerActor->currentPos.z);
+        partnerActor->shadowScale = partnerActor->size.x / 24.0;
+        partnerActor->hudElementDataIndex = create_status_icon_set();
+        partnerActor->ptrDefuffIcon = playFX_41(0, -142.0f, 34.0f, 1.0f, 0);
+        partnerActor->unk_228 = NULL;
+
+        takeTurnScript = start_script(partnerActor->takeTurnScriptSource, 10, 0);
+        partnerActor->takeTurnID = takeTurnScript->id;
+        takeTurnScript->owner1.actorID = ACTOR_PARTNER;
+    }
+}
 
 INCLUDE_ASM(s32, "190B20", create_actor);
 
-INCLUDE_ASM(s32, "190B20", func_80265CE8);
+s32 func_80265CE8(u32* anim, s32 arg1) {
+    s32 ret;
+
+    if (anim == 0) {
+        return 0;
+    }
+
+    ret = 0;
+    while (*anim != NULL) {
+        if (*anim == 1) {
+            ret = anim[1];
+        }
+        if (*anim == arg1) {
+            ret = anim[1];
+            break;
+        }
+        anim += 2;
+    }
+    return ret;
+}
+
+s32 func_80265D44(s32 arg0) {
```

Comment:
This arg is a player animation ID, I'd name it as such and add a TODO for us to use the player animation enum once we sort it